packages/react/src/inbox/adapters/GetAllConversationsPresenter.ts

```diff
@@ -0,0 +1,18 @@
+import {
+  GetAllConversationsResult,
+  IGetConversationsPresenter,
+} from '@lens-protocol/domain/use-cases/inbox';
+import { invariant } from '@lens-protocol/shared-kernel';
+
+export class GetAllConversationsPresenter implements IGetConversationsPresenter {
+  private result: GetAllConversationsResult | null = null;
+
+  presentConversations(r: GetAllConversationsResult) {
+    this.result = r;
+  }
+
+  asResult(): GetAllConversationsResult {
+    invariant(this.result, 'Result should be present');
```

Comment:
This invariant surface the fact that the caller of the use case using this hook, must know when the use case is finished before calling this method.

If you notice elsewhere this is use more as a "I call it but don't know when it ends... it will end at some point".

Reply:
what can I do about that? if I don't include `invariant` then TS complains that `this.result` could be `null`

Reply:
replaced with `PromiseResultPresenter`